### PR TITLE
fix auc op

### PR DESCRIPTION
**Pull request by typhoonzero**
project: https://github.com/PaddlePaddle/Paddle/issues/12008

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -92,8 +92,10 @@ install(DIRECTORY ${PADDLE_PYTHON_PACKAGE_DIR}
     DESTINATION opt/paddle/share/wheels
 )
 
-find_program(PATCHELF_EXECUTABLE patchelf)
-if(NOT PATCHELF_EXECUTABLE)
-  message(FATAL_ERROR "patchelf not found, please install it.\n"
+if (NOT APPLE)
```
Comment (jacquesqiao) [inline python/CMakeLists.txt]: fixed in https://github.com/PaddlePaddle/Paddle/pull/12106

File: python/paddle/fluid/tests/unittests/test_auc_op.py [hunk 2]
```diff
@@ -24,7 +24,20 @@ def setUp(self):
         indices = np.random.randint(0, 2, (128, 2))
         labels = np.random.randint(0, 2, (128, 1))
         num_thresholds = 200
-        self.inputs = {'Out': pred, 'Indices': indices, 'Label': labels}
+        tp = np.zeros((num_thresholds, )).astype("int32")
```
Comment (jacquesqiao) [inline python/paddle/fluid/tests/unittests/test_auc_op.py]: int32 -> int64

File: paddle/fluid/operators/auc_op.h [hunk 3]
```diff
@@ -54,19 +58,10 @@ class AucKernel : public framework::OpKernel<T> {
...
-    true_positive.Resize({num_thresholds});
-    false_negative.Resize({num_thresholds});
-    true_negative.Resize({num_thresholds});
-    false_positive.Resize({num_thresholds});
-
-    int64_t* tp_data = true_positive.mutable_data<int64_t>(ctx.GetPlace());
-    int64_t* fn_data = false_negative.mutable_data<int64_t>(ctx.GetPlace());
-    int64_t* tn_data = true_negative.mutable_data<int64_t>(ctx.GetPlace());
-    int64_t* fp_data = false_positive.mutable_data<int64_t>(ctx.GetPlace());
+    int64_t* tp_data = true_positive->mutable_data<int64_t>(ctx.GetPlace());
```
Comment (jacquesqiao) [inline paddle/fluid/operators/auc_op.h]:
```cpp
int64_t* -> auto*
```

**Comment by jacquesqiao** [comment]
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/metric_op.py#L79

**Comment by jacquesqiao** [comment]
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/metric_op.py#L79
这里也需要改一下

**Review by jacquesqiao — DISMISSED** [review]
LGTM!

File: python/paddle/fluid/layers/metric_op.py [hunk 4]
```diff
@@ -76,7 +76,7 @@ def accuracy(input, label, k=1, correct=None, total=None):
     return acc_out
 
 
-def auc(input, label, curve='ROC', num_thresholds=200):
+def auc(input, label, curve='ROC', num_thresholds=200, k=1):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/metric_op.py]: Need doc for `k` in following `Args`

Comment (typhoonzero) [reply python/paddle/fluid/layers/metric_op.py]: Done

**Review by jacquesqiao — APPROVED** [review]
LGTM!

**Pull request merged**